src/main/java/com/shoyuanime/animeAnthology/service/AnimeManager.java

```diff
@@ -0,0 +1,60 @@
+package com.shoyuanime.animeAnthology.service;
+
+import com.shoyuanime.animeAnthology.dto.AnimeDTO;
+import com.shoyuanime.animeAnthology.mapper.AnimeMapper;
+import com.shoyuanime.animeAnthology.model.Anime;
+import com.shoyuanime.animeAnthology.repository.AnimeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@Service
+@Validated
+public class AnimeManager {
+
+    @Autowired
+    private AnimeRepository animeRepository;
+    @Autowired
+    private AnimeMapper mapper;
+
+    public Optional<AnimeDTO> createAnime(@Valid AnimeDTO animeDTO) {
+        if (animeDTO.getId() != null && animeRepository.existsById(animeDTO.getId())) {
+            return Optional.empty();
+        }
+        Anime newAnime = mapper.map(animeDTO);
+        return Optional.of(mapper.map(animeRepository.save(newAnime)));
+    }
+
+    public Optional<AnimeDTO> updateAnime(@Valid AnimeDTO animeDTO) {
+        Optional<Anime> existingAnime = animeRepository.findById(animeDTO.getId());
+        if (existingAnime.isPresent()) {
+            // levels.id isn't mapped, so we need to store it early
+            Long levelsId = existingAnime.get().getLevels().getId();
```

Comment:
What if the levels field was never populated in the first place? Or is it assumed that this will always be populated?